src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -113,28 +123,169 @@ export default class GroupComparisonStore {
     });
 
     private _mutationEnrichmentProfile:MolecularProfile|undefined = undefined;
-    @computed public get mutationEnrichmentProfile() {
-        if (!this._mutationEnrichmentProfile && this.mutationEnrichmentProfiles.isComplete) {
-            return this.mutationEnrichmentProfiles.result[0];
-        } else {
-            return this._mutationEnrichmentProfile;
+    readonly mutationEnrichmentProfile = remoteData({
+        await:()=>[this.mutationEnrichmentProfiles],
+        invoke:()=>{
+            if (!this._mutationEnrichmentProfile && this.mutationEnrichmentProfiles.result!.length > 0) {
+                return Promise.resolve(this.mutationEnrichmentProfiles.result![0]);
+            } else {
+                return Promise.resolve(this._mutationEnrichmentProfile);
+            }
         }
-    }
-    public set mutationEnrichmentProfile(profile:MolecularProfile|undefined) {
+    });
+    public setMutationEnrichmentProfile(profile:MolecularProfile|undefined) {
```

Comment:
@action?   i think it's good to keep up the convention even if when there's only one observable set it means nothing